src/components/nav/Nav.js

```diff
@@ -0,0 +1,39 @@
+import "./Nav.css";
+import React from "react";
+import { BrowserRouter as Router, Link } from "react-router-dom";
+import { useCart } from "../../contexts/Cart-Context";
+import { useWishlist } from "../../contexts/Wishlist-Context";
+import { CgShoppingCart, CgHeart, CgMediaPodcast } from "react-icons/cg";
+import { FcShop } from "react-icons/fc";
+
+export function Nav() {
+  const { cartState, cartDispatch } = useCart();
+  const { wishlistItems, setwishlistItems } = useWishlist();
+
+  return (
+    <div>
+      <nav className="nav1">
+        <div className="logo">
+          <img className="navLogo" src="/ecom-nav-favicon.png" />
+        </div>
+        <input className="navSearchBar" placeholder="🔎Search something" />
+        <div className="navIcons">
+          <Link className="routeLink" to="/">
+            <CgMediaPodcast />
+          </Link>
+          &nbsp;
+          <Link className="routeLink" to="/cart">
+            <span className="iconBadge">{cartState.cart.length}</span>
```

Comment:
Currently, in your UI - navbar's icon:\
if your data is 0 - the badge is still visible;\
what you can try is - when the data is 0 - make the badge to *display: none*

Reply:
Thanks, Added this feature